queue_services/entity-filer/src/entity_filer/filing_processors/correction.py

```diff
@@ -24,6 +24,32 @@
 from entity_filer.filing_processors.filing_components.correction import correct_business_data
 
 
+def is_special_resolution_correction(filing: Dict, business: Business, original_filing: Filing):
+    """Check whether it is a special resolution correction."""
+    corrected_filing_type = filing['correction']['correctedFilingType']
+    is_coop = business.legal_type in ['CP']
+
+    if not is_coop:
+        return False
+    
+    if corrected_filing_type == 'specialResolution':
+        return True
+    
+    # Check to see if top level original filing is a specialResolution
+    original_filing_correction = original_filing.filing_json['filing']['correction']
+    while original_filing_correction.get('correctedFilingType') == 'correction':
+        if (
+            corrected_filing_type == 'correction' and
+            original_filing_correction .get('correctedFilingType') == 'specialResolution'
+        ):
+            return True
+        # Find the next original filing in the chain of corrections
+        original_filing = Filing.find_by_id(original_filing_correction.get('correctedFilingId'))
+        original_filing_correction = original_filing.filing_json['filing']['correction']
+
+    return corrected_filing_type == 'specialResolution'
```

Comment:
already have this on line 35, should be return False

Reply:
I re-wrote this function